checkapi/internal/diff_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiff(t *testing.T) {
+	a, err := Read(filepath.Join("testpkg", "receiver", "validreceiver"), nil, nil)
+	require.NoError(t, err)
+	b := a
+	bPreConst := b
+	bPreConst.Values = append([]string{"foobar"}, bPreConst.Values...)
+	bAddConst := b
+	bAddConst.Values = append(bAddConst.Values, "foobar")
+	bAddFn := b
+	bAddFn.Functions = append(bAddFn.Functions, Function{
+		Name:        "MyFn",
+		Receiver:    "foo",
+		ReturnTypes: []string{"string", "error"},
+		Params:      []string{"string", "bool"},
+		TypeParams:  []string{"~string"},
+	})
+	bBeforeFn := b
+	bBeforeFn.Functions = append([]Function{{
+		Name:        "Aaaa",
+		Receiver:    "foo",
+		ReturnTypes: []string{"string", "error"},
+		Params:      []string{"string", "bool"},
+	}}, bBeforeFn.Functions...)
+	bAddStruct := b
+	bAddStruct.Structs = append(bAddStruct.Structs, Apistruct{Name: "MyStruct", Fields: []string{"foo", "bar"}})
+
+	tests := []struct {
```

Comment:
Consider adding the following test cases:

- adding another  type parameter to an existing function (this is a backwards compatible change)
- adding a new parameter to a function (this is a breaking change)

Reply:
I added both, but they probably don't do what you want. Right now, the algorithm just finds the api to be not equal. I think you are asking for a different treatment - where you'd get right breaking change, right, equal, left, left breaking change.